solutionbox/structured_data/mltoolbox/_structured_data/_package.py

```diff
@@ -59,15 +59,7 @@ class FileNotFoundError(IOError):
 
 def _default_project():
   from google.datalab import Context
-  return Context.default().project_id
-
-
-def _is_in_IPython():
-  try:
-    import IPython # noqa
-    return True
-  except ImportError:
-    return False
+  return Context.default().project_id.decode()
```

Comment:
py2: Context.default().project_id is type str, after decode() type is  unicode.
py3: Context.default().project_id is type bytes, after decode() type is str (which is unicode)

Moved this logic into get_default_project_id() function.

now:
in py2: project_id is type str
in py3: project_id is type str

The problem is that the output of  subprocess.Popen().communicate()  can be anything (say image data), so this isn't really a compatability question of converting str to unicode, but how bytes should be converted to a string type.